### PR TITLE
#3694 Replace api with context actions in components

### PR DESCRIPTION
**Pull request by saivirtue**
### Checklist
- [x] [Review Contribution Guidelines](https://ohara.readthedocs.io/en/latest/contributing.html)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#ohara-team))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #3694)
- [ ] Add unit tests (or please explain why this PR is not worth having new tests)
- [ ] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: ohara-manager/client/src/components/Workspace/Create/WorkspaceQuick.js [hunk 1]
```diff
@@ -135,23 +127,10 @@ const WorkspaceQuick = props => {
...
-
-  const onDrop = async (file, values) => {
-    const parentKey = { group: GROUP.workspace, name: values.workspaceName };
-    const result = await fileApi.create({
-      group: hashByGroupAndName(GROUP.workspace, values.workspaceName),
-      file: file[0],
-      tags: { parentKey },
-    });
+  const onDrop = async file => {
+    const result = await createFile(file[0]);
```
Comment (eechih) [inline ohara-manager/client/src/components/Workspace/Create/WorkspaceQuick.js]: @saivirtue  在 Create workspace 的階段下，ApiContext fileApi 的 fileGroup 會是 null 或指到上一個 workspace，所以直接使用 fileApi 應該會出現錯誤... 🤔

File: ohara-manager/client/src/components/Workspace/WorkspaceList.js [hunk 2]
```diff
@@ -59,19 +61,25 @@ Statistic.propTypes = {
...
     const fetchTopics = async () => {
       const result = await topicApi.getAll();
       setTopics(result.errors ? [] : result.data);
     };
-    if (topics !== null) return;
-    fetchTopics();
-  }, [topics]);
+    // if we add some topic into current workspace
+    // re-fetch all topics again
+    if (topics === null || prevTopics !== currentTopics) {
```
Comment (konekoya) [inline ohara-manager/client/src/components/Workspace/WorkspaceList.js]: No a bit deal, but we could move this check to the top of this effect and so we can return early if these conditions are met :)

Comment (konekoya) [reply ohara-manager/client/src/components/Workspace/WorkspaceList.js]: E.g.,

```
React.useEffect(() => {

if (...conditions) return;

const fetchTopics = async => { ... function body }


// rest of our logic

}, [deps...])
```

File: ohara-manager/client/src/components/Workspace/Create/WorkspaceQuick.js (hunk 1 above)
Comment (saivirtue) [reply ohara-manager/client/src/components/Workspace/Create/WorkspaceQuick.js]: You are right :(
After all, we still have some exceptions that context action could not be used.

File: ohara-manager/client/src/components/Workspace/WorkspaceList.js (hunk 2 above)
Comment (saivirtue) [reply ohara-manager/client/src/components/Workspace/WorkspaceList.js]: These checks are already in the top of useEffect :)
```javascript
React.useEffect(() => {
   const fetchTopics = ....;
   if(conditions) {
      fetchTopics();
   }
});
```

Comment (konekoya) [reply ohara-manager/client/src/components/Workspace/WorkspaceList.js]: Oops, I must have missed them

**Pull request merged**